enterprise/utils.py

```diff
@@ -123,3 +124,28 @@ def patch_mako_lookup():
     # Both add an item to the setting AND insert the lookup for immediate use
     settings.MAKO_TEMPLATES['main'].insert(0, template_location)
     add_lookup('main', template_location)
+
+
+def get_catalog_admin_url_template():
+    """
+    Get template of catalog admin url.
+
+    URL template will contain a placeholder '{catalog_id}' for catalog id.
+
+    Example:
+        >>> get_catalog_admin_url_template()
+        "http://localhost:18381/admin/catalogs/catalog/{catalog_id}/change/"
+
+    Returns:
+        A string containing template for catalog url.
+    """
+    api_base_url = getattr(settings, "COURSE_CATALOG_API_URL", "")
+
+    # Extract FQDN (Fully Qualified Domain Name) from API URL.
+    match = re.match(r"^(?P<fqdn>(?:https?://)?[^/]+)", api_base_url)
```

Comment:
@mattdrayer I tried using `urllib` module but it expects urls with proper scheme i.e. 'http', 'https' etc. and urls like 'localhost:18381/api/v1' return invalid result.
```
>>> from urllib.parse import urlparse
>>>  # Scheme is 'localhost' here it should be empty. I have printed another result for comparison
>>> urlparse("localhost:18387/api/v1/")
ParseResult(scheme='localhost', netloc='', path='18387/api/v1/', params='', query='', fragment='')
>>> urlparse("https://discovery-site.subdomain.example.com/api/v1/")
ParseResult(scheme='https', netloc='discovery-site.subdomain.example.com', path='/api/v1/', params='', query='', fragment='')
```